contracts/FastEcMul.sol

```diff
@@ -443,7 +443,7 @@ library FastEcMul {
   /// @param _x the integer
   /// @return y the square root of _x
   function _sqrt(uint256 _x) private pure returns (uint256) {
-    uint256 z = (_x + 1) / 2;
+    uint256 z = _x / 2;
```

Comment:
This fails when `_x` is 1, returning 0 instead of 1. A simple fix would be to add a check for that case:

```suggestion
    if(x == 1) return 1;
    uint256 z = _x / 2;
```